lib/PSCommand.js

```diff
@@ -0,0 +1,69 @@
+const { getType, convertToPSParam } = require('./utils');
+const {
+  PS_ARG_MISS_ERROR,
+  PS_ARG_TYPE_ERROR,
+} = require('./errors');
+
+class PSCommand {
+  constructor(command) {
+    if(!command) {
+      throw new PS_ARG_MISS_ERROR('Command is missing');
+    }
+    if(getType(command) !== 'String') {
+      throw new PS_ARG_TYPE_ERROR('Command must be a string');
+    }
+
+    this.command = command;
+  }
+  addArgument(argument = '') {
+    if(!argument) {
+      throw new PS_ARG_MISS_ERROR('Argument is missing');
+    }
+    if(getType(argument) !== 'String') {
+      throw new PS_ARG_TYPE_ERROR('Argument must be a string');
+    }
+
+    this.command = `${this.command} ${argument}`;
+    return this;
```

Comment:
I'm not sure I'd recommend a mutating builder API - I'm wondering if it doesn't make more sense to do a `return new this.constructor(`${this.command} ${argument}`)`

Reply:
explanation why its better :)

Reply:
because it's surprising that a fluid interface mutates the underlying object